src/lib.rs

```diff
@@ -248,3 +258,33 @@ impl Debug {
         obj.set_colour(Colour::Black)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_trait_obj() {
+        trait HelloWorld {
+            fn hello(&self) -> usize;
+        }
+
+        struct HelloWorldStruct(usize);
+
+        impl HelloWorld for HelloWorldStruct {
+            fn hello(&self) -> usize {
+                self.0
+            }
+        }
+
+        let s = HelloWorldStruct(123);
+        let gcto: Gc<dyn HelloWorld> = Gc::new(s);
```

Comment:
I think you probably need two `impl`s for this struct, otherwise Rust might end up turning this into static dispatch. Alternatively, you could assert that `size_of::<Gc<dyn HelloWorld>>()` is `2 * usize`?

Reply:
> Alternatively, you could assert that size_of::<Gc<dyn HelloWorld>>() is 2 * usize?

Good idea. I'll add this.

Reply:
Fixed: 75d151976e9ab0f6b8883c38c719450b3978877c